src/Sylius/Bundle/CoreBundle/EventListener/UserDeleteListener.php

```diff
@@ -42,14 +42,14 @@ public function deleteUser(ResourceEvent $event)
     {
         $user = $event->getSubject();
 
-        if (!$user instanceof UserInterface) {
+        if (!$user instanceof GroupInterface) {
             throw new UnexpectedTypeException(
                 $user,
-                'FOS\UserBundle\Model\UserInterface'
+                'FOS\UserBundle\Model\GroupInterface'
             );
         }
 
-        if ($this->securityContext->getToken()->getUsername() === $user->getUsernameCanonical()) {
+        if ($this->securityContext->getToken()->getUser()->getId() === $user->getId()) {
```

Comment:
This will die will fatal error if user got somehow signed-off, as token would return `null` not user object...